### PR TITLE
introduce value objects in transfer

### PR DESCRIPTION
**Pull request by martinhenningjensen**

### CODE REVIEW

File: domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
     [Fact]
     public void Tin_WithValidData_CreatesSuccessfully()
     {
-        var tin = new Tin("00000000");
+        var tin = Tin.Create("00000000");
         Assert.Equal("00000000", tin.Value);
     }
 
     [Fact]
     public void Tin_WithInvalidLength_ThrowsException()
```
Comment (tnickelsen) [inline domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs:15]: Minor: Make Theory and add test for less than 8 digits

**Review by tnickelsen — APPROVED** [review]

File: domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs (hunk 1 above)
Comment (martinhenningjensen) [reply domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs:14]: Added test test for less than 8 digits, the Theory part, do you mean to collapse all the tests into one With Theory cases?

Comment (tnickelsen) [reply domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs:14]: Nah just this test case

Comment (martinhenningjensen) [reply domains/libraries/EnergyOrigin.Domain/EnergyOrigin.Domain.ValueObjects.Tests/TinTest.cs]: Done

**Pull request merged**